test/test_model.py

```diff
@@ -3816,6 +3816,62 @@ def test_set_info(self, model: ops.Model, fake_script: FakeScript):
         with pytest.raises(TypeError):
             secret.set_info()  # no args provided
 
+    def test_set_content_then_info(self, model: ops.Model, fake_script: FakeScript):
+        fake_script.write('secret-set', """exit 0""")
+        fake_script.write('secret-get', """echo '{"foo": "bar"}'""")
+
+        secret = self.make_secret(model, id='q')
+        secret.set_content({'foo': 'bar'})
+        description = 'desc'
+        secret.set_info(description=description)
+
+        calls = fake_script.calls(clear=True)
+        assert calls == [
+            ['secret-set', f'secret://{model._backend.model_uuid}/q', ANY],
+            ['secret-get', f'secret://{model._backend.model_uuid}/q', '--peek', '--format=json'],
+            [
+                'secret-set',
+                f'secret://{model._backend.model_uuid}/q',
+                '--description',
+                description,
+                ANY,
```

Comment:
Huh, TIL you can use `ANY` in normal `assert` statements / equality comparisons. I would have thought it was only for `assert_called_with` and co. Cool!

Reply:
It is a little ugly that this file now does:
```
from unittest.mock import ANY, MagicMock, patch
```

and
```
from typing import (
    Any,
```

So you have both an ANY and an Any and they aren't the same thing nor can be used in the same way.

https://docs.python.org/3/library/unittest.mock.html#any

I guess the implementation is just a type who always returns True for any equality test:
https://github.com/python/cpython/blob/c7a94e77bc61ed6a1ec1419748f2eaefa67e0c26/Lib/unittest/mock.py#L2462-L2474

Reply:
For what it's worth, the introduction of `unittest.mock.ANY` was done a couple of months ago - I was just following the previous pattern. I would have personally had them both living in their namespaces so that it was `typing.Any` and `mock.ANY` (or maybe `unittest.mock.ANY`).

I'm fine reworking the tests to not use `ANY` (it does seem a bit odd to be using a `unittest.mock` object when we're not actually doing the mocks with `unittest.mock`), for example:

```python
assert calls[0][:-1] == ['secret-set', f'secret://{model._backend.model_uuid}/q', ANY]
assert calls[1] == ['secret-get', f'secret://{model._backend.model_uuid}/q', '--peek', '--format=json']
assert calls[2] == [
    'secret-set',
    f'secret://{model._backend.model_uuid}/q',
    '--description',
    description,
    ANY,
]
```

But presumably I should then adjust the existing tests as well? Or I could change the import and adjust based on that.

Reply:
Let's do `from unittest import mock` so we get `mock.ANY` and the like -- probably best to do that in a follow-up PR for cleanliness if you prefer.

Reply:
Opened #1379, and I'll adjust this one after that gets merged (which I assume will be quicker than this one).